rules/ruleset.go

```diff
@@ -502,10 +516,20 @@ func NewRuleSet(version int, rs *schema.RuleSet, opts Options) (RuleSet, error)
 	}, nil
 }
 
+// NewRuleSet creates a blank concrete ruleSet
+func NewRuleSet() RuleSet {
+	return &ruleSet{}
+}
+
+func (rs *ruleSet) Handler(opts HandlerOptions) RuleSetHandler {
+	//Todo(dgromov): This should create a copy.
```

Comment:
What's the implication of not making a copy? would it corrupt the data if you have two operations going on at the same time? If so I'd rather do it in this PR.